miden/src/cli/data.rs

```diff
@@ -164,7 +165,11 @@ pub struct ProgramFile;
 
 /// Helper methods to interact with masm program file
 impl ProgramFile {
-    pub fn read(path: &PathBuf, debug: bool) -> Result<Program, String> {
+    pub fn read(
+        path: &PathBuf,
+        debug: bool,
+        library: &Option<MaslLibrary>,
```

Comment:
I would change this to `&[MaslLibrary]`. This way, when we allow specifying multiple library files no extra changes would be needed here.

Reply:
I suggest we go even more generic there and put an `IntoIterator<Item = L>` where `L Borrow: MaslLibrary` so we can cover also slices.